course7/homeworks/mathias/complex_rpn/complex.cpp

```diff
@@ -0,0 +1,61 @@
+#include "complex.h"
+
+#include <cmath>
+
+Complex::Complex(double real, double imaginary)
+	: mReal(real)
+	, mImaginary(imaginary)
+{ }
+
+Complex::Complex(const std::string & str)
+{
+	size_t index = 0;
+
+	try
+	{
+		mReal = std::stod(str, &index);
+
+		if (str.substr(index, 2) == "+i")
+		{
+			mImaginary = std::stod(str.substr(index + 2));
+		}
+	}
+	catch (const std::invalid_argument&)
+	{
+		throw std::runtime_error("Input could not be convert to a complex number.");
+	}
+	catch (const std::out_of_range&)
+	{
+		throw std::runtime_error("Input value too big to store.");
+	}
+}
+
+Complex Complex::operator+(const Complex& rhs)
+{
+	return Complex { mReal + rhs.mReal, mImaginary + rhs.mImaginary };
+}
+
+Complex Complex::operator-(const Complex& rhs)
```

Comment:
It is better to place functions this size in the header file, so that compiler can see them and inline them when compiling other compilation units.